### PR TITLE
Bump gtk to 0.16

### PR DESCRIPTION
**Pull request by pentamassiv**
Thank you for writing `relm`.
A new version of gtk and their related crates was released. This PR allows using it. The minimal Rust version is now 1.63.0 (due to gio). There are two caveats though, but they only apply to some of the examples:

- The webkit-test example no longer works and was disabled in the Github Workflow. The reason for this is that webkit2gtk needs to publish a new version which is compatible with gtk 0.16.
- The [gladis crate](https://crates.io/crates/gladis) was deprecated so there will not be a version that is compatible with gtk 0.16. I removed the gladis example. The replacement for this crate are [CompositeTemplate](https://gtk-rs.org/gtk4-rs/stable/latest/book/composite_templates.html) from [gtk3-macros](https://crates.io/crates/gtk3-macros), but I did not write an example with that.

I also updated a few other dependencies in the examples so they should all be up-to-date now :)

BTW, the code could use a `cargo fmt`. I would have done it, but it would probably be easier for you to run it so you don't need to review the changes.

### CODE REVIEW

**Comment by pentamassiv** [comment]
I reverted the formatting changes and added a rustfmt.toml file so others don't try to auto-format the code. The PR is ready for your review :)

File: src/core/mod.rs [hunk 1]
```diff
@@ -264,7 +265,7 @@ pub struct EventStream<MSG> {
 impl<MSG> Drop for EventStream<MSG> {
     fn drop(&mut self) {
         // Ignore error since we're in a destructor.
```
Comment (antoyo) [inline src/core/mod.rs:267]: Please remove this comment as it is not useful anymore.

**Review by antoyo — COMMENTED** [review]
I forgot this.
After that, it will be good.
Thanks.

File: src/core/mod.rs (hunk 1 above)
Comment (pentamassiv) [reply src/core/mod.rs:267]: Sure

Comment (antoyo) [reply src/core/mod.rs:267]: I see you marked this issue as resolved, but that comment was not removed. Did you forget to push the changes?

**Comment by pentamassiv** [comment]
Sorry, I committed it on the wrong branch. Now it is gone

File: src/core/mod.rs (hunk 1 above)
Comment (pentamassiv) [reply src/core/mod.rs]: Sorry, I committed it to the wrong branch. Now it is gone

**Review by antoyo — APPROVED** [review]
Thanks a lot!

**Pull request merged**